assets/js/googlesitekit/datastore/modules/modules.js

```diff
@@ -0,0 +1,434 @@
+	 *
+	 * Activate a module (based on the slug provided).
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string} slug Slug of the module to activate.
+	 * @return {Object}      Object with {response, error}
+	 */
+	*activateModule( slug ) {
+		const { response, error } = yield actions.setModuleStatus( slug, true );
+		return { response, error };
+	},
+
+	/**
+	 * Dectivate a module on the server.
+	 *
+	 * Dectivate a module (based on the slug provided).
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string} slug Slug of the module to activate.
+	 * @return {Object}      Object with {response, error}
+	 */
+	*deactivateModule( slug ) {
+		const { response, error } = yield actions.setModuleStatus( slug, false );
+		return { response, error };
+	},
+
+	/**
+	 * (De)activate a module on the server.
+	 *
+	 * POSTs to the `core/modules/activation` endpoint to set the `active` status
+	 * supplied for the give `slug`.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param  {string}  slug   Slug of the module to activate/deactivate.
+	 * @param  {boolean} active `true` to activate; `false` to deactivate.
+	 * @return {Object}         Object with {response, error}
+	 */
+	*setModuleStatus( slug, active ) {
+		invariant( slug, 'slug is required.' );
+		invariant( active !== undefined, 'active is required.' );
+
+		let response, error;
+
+		yield {
+			payload: { active, slug },
+			type: START_FETCH_SET_MODULE_STATUS,
+		};
+
+		try {
+			response = yield {
+				payload: { active, slug },
+				type: FETCH_SET_MODULE_STATUS,
+			};
+
+			if ( response.success === true ) {
+				// Fetch (or re-fetch) all modules, with their updated status.
+				yield actions.fetchModules();
+			}
+
+			yield {
+				payload: { active, slug },
+				type: FINISH_FETCH_SET_MODULE_STATUS,
+			};
+		} catch ( e ) {
+			error = e;
+			yield {
+				payload: { active, slug, error },
+				type: CATCH_FETCH_SET_MODULE_STATUS,
+			};
+		}
+
+		return { response, error };
+	},
+
+	/**
+	 * Dispatches an action that creates an HTTP request.
+	 *
+	 * Requests the `core/modules/list` endpoint.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @return {Object} Object with {response, error}
+	 */
+	*fetchModules() {
+		let response, error;
+
+		yield {
+			payload: {},
+			type: START_FETCH_MODULES,
+		};
+
+		try {
+			response = yield {
+				payload: {},
+				type: FETCH_MODULES,
+			};
+
+			yield actions.receiveModules( response );
+
+			yield {
+				payload: {},
+				type: FINISH_FETCH_MODULES,
+			};
+		} catch ( e ) {
+			error = e;
+			yield {
+				payload: { error },
+				type: CATCH_FETCH_MODULES,
+			};
+		}
+
+		return { response, error };
+	},
+
+	/**
+	 * Receive the API respond for module (de)activation.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param  {string}  slug    Name of slug to activate/deactivate.
+	 * @param  {boolean} active  `true` to activate; `false` to deactivate.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveSetModuleStatus( slug, active ) {
```

Comment:
Currently unused since the new module data is updated via `fetchModules` instead. So let's remove it then.